src/readers/earnings.ts

```diff
@@ -63,40 +38,34 @@ export class EarningsReader<C extends ChainId> extends Reader<C> {
   }
 
   async assetEarnings(vaultAddress: Address): Promise<AssetEarnings> {
-    interface VaultDataContainer {
-      vault: Vault;
-    }
-
-    const query = `
-    {
-      vault(id: "${vaultAddress.toLowerCase()}") {
-        token {
-          id
-          decimals
-        }
-        latestUpdate {
-          returnsGenerated
-        }
+    const response = await this.yearn.services.subgraph.client.query<VaultEarnings, VaultEarningsVariables>({
+      query: VAULT_EARNINGS,
+      variables: {
+        vault: vaultAddress
       }
+    });
+
+    const vault = response.data.vault;
+
+    if (vault == undefined) {
```

Comment:
```suggestion
    if (vault === undefined) {
```

https://eslint.org/docs/rules/eqeqeq
or even:

```suggestion
    if (!vault) {
```

Reply:
Thanks, still learning ts